### PR TITLE
BEES enhancements & features

### PR DESCRIPTION
**Pull request by ghost**
Added lots of new features & enhancements to BEES, some new ops & also compiled all BEES ops (including grid ops) into puredata externals.  The focus-switching should also work within PD.

All these issues are fixed or partially fixed:

https://github.com/monome/aleph/issues/278
https://github.com/monome/aleph/issues/277
https://github.com/monome/aleph/issues/280
https://github.com/monome/aleph/issues/281
https://github.com/monome/aleph/issues/276
https://github.com/monome/aleph/issues/253
https://github.com/monome/aleph/issues/169
https://github.com/monome/aleph/issues/251

Also focus-swapping, per-instance op state allocation & pickling for meadowphysics, whitewhale, life (i.e you can fire up multiple of these now & save to a scene)

However, ran into something I absolutely can not comprehend or figure out how to debug:
https://github.com/rick-monster/aleph/issues/27
In a nutshell if you try to swap module on a scene containing certain ops (e.g screen, whitewhale) then BEES will explode when you try to access those certain ops.

Don't think I can crack this on my own - just banged my head against it for literally 10 hours getting nowhere... seems to be a very tough problem...  Anyone able to suggest a way forward?

### CODE REVIEW

**Comment by ghost** [comment]
Ah hell, the bug was simple - my big/small opPool code blew the memory budget in an unexpected way.  So the 64k alloc for blackfin prog starting at byte 991972(decimal) was exploding for obvious reasons.  Didn't see this before, because many of the ops weren't 'bigOps' - the focus swapping code increased size of all grid ops (by giving them their own LED buffer)

Fixed by reducing max. number of bigOps in mempool, no longer seeing scary/worrying crashes...

**Comment by ngwese** [comment]
which kria did you pull from ww or ansible?

**Comment by ghost** [comment]
ww

**Comment by ghost** [comment]
Reopening the meat of this pull-request from a different branch (hacking a bit too frantically on dev to let this PR stand).
